### ISSUE TITLE
Backend TLS: Gateway to Backend tutorial is broken

### ISSUE THREAD

**Issue by whitneygriffith** [envoyproxy/gateway#4019]
*Description*:
>What issue is being seen? Describe what should be happening instead of
the bug, for example: Envoy should not crash, the expected value isn't
returned, etc.

I followed [the tutorial ](https://gateway.envoyproxy.io/latest/tasks/security/backend-tls/)to test BackendTLSPolicy with Envoy Gateway.

And I got the following error.

```
http://www.example.com:80/get
* Added www.example.com:80:172.19.0.0 to DNS cache
* Hostname www.example.com was found in DNS cache
*  Trying 172.19.0.0:80...
* Connected to www.example.com (172.19.0.0) port 80 (#0)
> GET /get HTTP/1.1
> Host:www.example.com
> User-Agent: curl/7.81.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 503 Service Unavailable
< content-length: 216
< content-type: text/plain
< date: Wed, 07 Aug 2024 18:33:25 GMT
<
* Connection #0 to host www.example.com left intact
upstream connect error or disconnect/reset before headers. reset reason: remote connection failure, transport failure reason: TLS_error
```

Some things I have tried:

- I confirmed my secret and configmap matches the certs I created locally 

- Checked Envoy logs which showed the same TLS error `OPENSSL_internal:CERTIFICATE_VERIFY_FAILED:TLS_error_end`

`{"start_time":"2024-08-07T18:33:25.449Z","method":"GET","x-envoy-origin-path":"/get","protocol":"HTTP/1.1","response_code":"503","response_flags":"UF","response_code_details":"upstream_reset_before_response_started{remote_connection_failure|TLS_error:|268435581:SSL_routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED:TLS_error_end}","connection_termination_details":"-","upstream_transport_failure_reason":"TLS_error:|268435581:SSL_routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED:TLS_error_end","bytes_received":"0","bytes_sent":"216","duration":"2","x-envoy-upstream-service-time":"-","x-forwarded-for":"172.19.0.1","user-agent":"curl/7.81.0","x-request-id":"330d3eeb-be79-48e0-816d-e1f2e76f3741",":authority":"www.example.com","upstream_host":"10.244.0.10:8443","upstream_cluster":"httproute/default/backend/rule/0","upstream_local_address":"-","downstream_local_address":"10.244.0.9:10080","downstream_remote_address":"172.19.0.1:35026","requested_server_name":"-","route_name":"httproute/default/backend/rule/0/match/0/www_example_com"}
`

TLDR, I am not sure why this tutorial does not work out of the box. The call being made is

```
export GATEWAY_HOST=$(kubectl get gateway/eg -o jsonpath='{.status.addresses[0].value}')
curl -v -HHost:www.example.com --resolve "www.example.com:80:${GATEWAY_HOST}" \
http://www.example.com:80
```

The backend deployment is configured to pull the correct secret from the k8s secret I created to store it

```
Plain Text
- name: secret-volume
    secret:
      secretName: example-cert
      items:
      - key: tls.crt
        path: crt
      - key: tls.key
        path: key
```

The backend pod is using the `gcr.io/k8s-staging-gateway-api/echo-basic:v20231214-v1.0.0-140-gf544a46e` image. Does anyone know If the echo basic image webserver is able to read from the secret-volume and displaying a certificate when it’s accessed?

Any other ideas of what may be wrong?
